packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -779,10 +779,14 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
       (widget.route.navigator?.userGestureInProgress ?? false);
   }
 
+  bool get _shouldTakeFocus {
```

Comment:
I could merge this with `_shoudlIgnoreFocusRequest` above. I decided not to because this function is doing something fairly distinct. Either seems fine to me though.

Reply:
I agree. I would call this `_shouldRequestFocus`, though.